From baf90b0045eff45d61f0786418ebb02eeb60022d Mon Sep 17 00:00:00 2001
From: "chengzhi.hou" <chengzhi.hou@ck-telecom.com>
Date: Wed, 24 Feb 2016 20:01:17 +0800
Subject: [PATCH 04/21] Messaging: Swipe right to delete conversation

Change "swipe delete" to "swipe right delete", so SwipeArchive and SwipeDelete
can exist at the same time.

Change-Id: Id66167c2e749f8ab0ee917e0138027c7404b8c90
---
 res/values/cm_strings.xml                             |  2 +-
 res/values/lineage_constants.xml                      |  4 ++--
 res/xml-v21/preferences_application.xml               |  4 ++--
 res/xml-v23/preferences_application.xml               |  4 ++--
 res/xml/preferences_application.xml                   |  4 ++--
 .../ui/appsettings/ApplicationSettingsActivity.java   | 11 ++++++-----
 .../ui/conversationlist/ConversationListItemView.java | 11 +++++++----
 .../conversationlist/ConversationListSwipeHelper.java |  8 ++++----
 src/org/lineageos/messaging/util/PrefsUtils.java      |  6 +++---
 9 files changed, 29 insertions(+), 25 deletions(-)

diff --git a/res/values/cm_strings.xml b/res/values/cm_strings.xml
index 7759e77..09c6d80 100644
--- a/res/values/cm_strings.xml
+++ b/res/values/cm_strings.xml
@@ -15,5 +15,5 @@
     limitations under the License.
 -->
 <resources xmlns:xliff="urn:oasis:names:tc:xliff:document:1.2">
-    <string name="swipe_deletes_conversation_text">Swipe deletes conversation</string>
+    <string name="swipe_right_deletes_conversation_text">Swipe right deletes conversation</string>
 </resources>
diff --git a/res/values/lineage_constants.xml b/res/values/lineage_constants.xml
index a6c92e1..1975b6a 100644
--- a/res/values/lineage_constants.xml
+++ b/res/values/lineage_constants.xml
@@ -17,8 +17,8 @@
 <resources>
     <!-- Preference keys for user-visible settings -->
     <!-- Application-wide settings -->
-    <string name="swipe_deletes_conversation_key" translatable="false">swipe_deletes_conversation</string>
+    <string name="swipe_right_deletes_conversation_key" translatable="false">swipe_right_deletes_conversation</string>
 
     <!-- This should really go into a config xml, but whoever wrote this app is an idiot, so follow their pattern -->
-    <bool name="swipe_deletes_conversation_default" translatable="false">false</bool>
+    <bool name="swipe_right_deletes_conversation_default" translatable="false">false</bool>
 </resources>
diff --git a/res/xml-v21/preferences_application.xml b/res/xml-v21/preferences_application.xml
index 374b1d2..54ad1a3 100644
--- a/res/xml-v21/preferences_application.xml
+++ b/res/xml-v21/preferences_application.xml
@@ -67,8 +67,8 @@
         android:dependency="@string/notifications_enabled_pref_key" />
 
     <SwitchPreference
-        android:key="@string/swipe_deletes_conversation_key"
-        android:title="@string/swipe_deletes_conversation_text"
+        android:key="@string/swipe_right_deletes_conversation_key"
+        android:title="@string/swipe_right_deletes_conversation_text"
         android:defaultValue="false"
         android:persistent="true" />
 
diff --git a/res/xml-v23/preferences_application.xml b/res/xml-v23/preferences_application.xml
index c6b934a..6b3b870 100644
--- a/res/xml-v23/preferences_application.xml
+++ b/res/xml-v23/preferences_application.xml
@@ -68,8 +68,8 @@
         android:dependency="@string/notifications_enabled_pref_key" />
 
     <SwitchPreference
-        android:key="@string/swipe_deletes_conversation_key"
-        android:title="@string/swipe_deletes_conversation_text"
+        android:key="@string/swipe_right_deletes_conversation_key"
+        android:title="@string/swipe_right_deletes_conversation_text"
         android:defaultValue="false"
         android:persistent="true" />
 
diff --git a/res/xml/preferences_application.xml b/res/xml/preferences_application.xml
index f9895d0..b998695 100644
--- a/res/xml/preferences_application.xml
+++ b/res/xml/preferences_application.xml
@@ -67,8 +67,8 @@
         android:dependency="@string/notifications_enabled_pref_key" />
 
     <SwitchPreference
-        android:key="@string/swipe_deletes_conversation_key"
-        android:title="@string/swipe_deletes_conversation_text"
+        android:key="@string/swipe_right_deletes_conversation_key"
+        android:title="@string/swipe_right_deletes_conversation_text"
         android:defaultValue="false"
         android:persistent="true" />
 
diff --git a/src/com/android/messaging/ui/appsettings/ApplicationSettingsActivity.java b/src/com/android/messaging/ui/appsettings/ApplicationSettingsActivity.java
index d41c0b4..20ffebf 100644
--- a/src/com/android/messaging/ui/appsettings/ApplicationSettingsActivity.java
+++ b/src/com/android/messaging/ui/appsettings/ApplicationSettingsActivity.java
@@ -98,8 +98,8 @@ public class ApplicationSettingsActivity extends BugleActionBarActivity {
         private String mSmsEnabledPrefKey;
         private Preference mSmsEnabledPreference;
         private boolean mIsSmsPreferenceClicked;
-        private String mSwipeToDeleteConversationkey;
-        private SwitchPreference mSwipeToDeleteConversationPreference;
+        private String mSwipeRightToDeleteConversationkey;
+        private SwitchPreference mSwipeRightToDeleteConversationPreference;
 
         public ApplicationSettingsFragment() {
             // Required empty constructor
@@ -124,9 +124,10 @@ public class ApplicationSettingsActivity extends BugleActionBarActivity {
             mSmsDisabledPreference = findPreference(mSmsDisabledPrefKey);
             mSmsEnabledPrefKey = getString(R.string.sms_enabled_pref_key);
             mSmsEnabledPreference = findPreference(mSmsEnabledPrefKey);
-            mSwipeToDeleteConversationkey = getString(R.string.swipe_deletes_conversation_key);
-            mSwipeToDeleteConversationPreference =
-                    (SwitchPreference) findPreference(mSwipeToDeleteConversationkey);
+            mSwipeRightToDeleteConversationkey = getString(
+                    R.string.swipe_right_deletes_conversation_key);
+            mSwipeRightToDeleteConversationPreference =
+                    (SwitchPreference) findPreference(mSwipeRightToDeleteConversationkey);
             mIsSmsPreferenceClicked = false;
 
             final SharedPreferences prefs = getPreferenceScreen().getSharedPreferences();
diff --git a/src/com/android/messaging/ui/conversationlist/ConversationListItemView.java b/src/com/android/messaging/ui/conversationlist/ConversationListItemView.java
index af4394e..0e95c65 100644
--- a/src/com/android/messaging/ui/conversationlist/ConversationListItemView.java
+++ b/src/com/android/messaging/ui/conversationlist/ConversationListItemView.java
@@ -76,6 +76,8 @@ public class ConversationListItemView extends FrameLayout implements OnClickList
     private static String sPlusOneString;
     private static String sPlusNString;
 
+    private static final int SWIPE_DIRECTION_RIGHT = 2;
+
     public interface HostInterface {
         boolean isConversationSelected(final String conversationId);
         void onConversationClicked(final ConversationListItemData conversationListItemData,
@@ -504,11 +506,11 @@ public class ConversationListItemView extends FrameLayout implements OnClickList
         final int notificationBellVisiblity = mData.getNotificationEnabled() ? GONE : VISIBLE;
         mNotificationBellView.setVisibility(notificationBellVisiblity);
 
-        if (PrefsUtils.isSwipeToDeleteEnabled()) {
+        if (PrefsUtils.isSwipeRightToDeleteEnabled()) {
             mCrossSwipeArchiveLeftImageView.setImageDrawable(getResources()
                     .getDrawable(R.drawable.ic_delete_small_dark));
             mCrossSwipeArchiveRightImageView.setImageDrawable(getResources()
-                    .getDrawable(R.drawable.ic_delete_small_dark));
+                    .getDrawable(R.drawable.ic_archive_small_dark));
         } else {
             mCrossSwipeArchiveLeftImageView.setImageDrawable(getResources()
                     .getDrawable(R.drawable.ic_archive_small_dark));
@@ -548,9 +550,10 @@ public class ConversationListItemView extends FrameLayout implements OnClickList
         }
     }
 
-    public void onSwipeComplete() {
+    public void onSwipeComplete(int swipeDirection) {
         final String conversationId = mData.getConversationId();
-        if (PrefsUtils.isSwipeToDeleteEnabled()) {
+        if (PrefsUtils.isSwipeRightToDeleteEnabled()
+                && swipeDirection == ConversationListSwipeHelper.SWIPE_DIRECTION_RIGHT) {
             mData.deleteConversation();
             UiUtils.showToastAtBottom(R.string.conversation_deleted);
             return;
diff --git a/src/com/android/messaging/ui/conversationlist/ConversationListSwipeHelper.java b/src/com/android/messaging/ui/conversationlist/ConversationListSwipeHelper.java
index 4988259..ac2aeb0 100644
--- a/src/com/android/messaging/ui/conversationlist/ConversationListSwipeHelper.java
+++ b/src/com/android/messaging/ui/conversationlist/ConversationListSwipeHelper.java
@@ -44,9 +44,9 @@ public class ConversationListSwipeHelper implements OnItemTouchListener {
     private static final float PERCENTAGE_OF_WIDTH_TO_DISMISS = 0.4f;
     private static final float FLING_PERCENTAGE_OF_WIDTH_TO_DISMISS = 0.05f;
 
-    private static final int SWIPE_DIRECTION_NONE = 0;
-    private static final int SWIPE_DIRECTION_LEFT = 1;
-    private static final int SWIPE_DIRECTION_RIGHT = 2;
+    public static final int SWIPE_DIRECTION_NONE = 0;
+    public static final int SWIPE_DIRECTION_LEFT = 1;
+    public static final int SWIPE_DIRECTION_RIGHT = 2;
 
     private final RecyclerView mRecyclerView;
     private final long mDefaultRestoreAnimationDuration;
@@ -269,7 +269,7 @@ public class ConversationListSwipeHelper implements OnItemTouchListener {
     private void onSwipeGestureEnd(final ConversationListItemView itemView,
             final int swipeDirection) {
         if (swipeDirection == SWIPE_DIRECTION_RIGHT || swipeDirection == SWIPE_DIRECTION_LEFT) {
-            itemView.onSwipeComplete();
+            itemView.onSwipeComplete(swipeDirection);
         }
 
         // Balances out onSwipeGestureStart.
diff --git a/src/org/lineageos/messaging/util/PrefsUtils.java b/src/org/lineageos/messaging/util/PrefsUtils.java
index e4e781d..fa08af4 100644
--- a/src/org/lineageos/messaging/util/PrefsUtils.java
+++ b/src/org/lineageos/messaging/util/PrefsUtils.java
@@ -31,12 +31,12 @@ public class PrefsUtils {
      * the conversation rather than archiving it.
      * @return hopefully true
      */
-    public static boolean isSwipeToDeleteEnabled() {
+    public static boolean isSwipeRightToDeleteEnabled() {
         final BuglePrefs prefs = BuglePrefs.getApplicationPrefs();
         final Context context = Factory.get().getApplicationContext();
-        final String prefKey = context.getString(R.string.swipe_deletes_conversation_key);
+        final String prefKey = context.getString(R.string.swipe_right_deletes_conversation_key);
         final boolean defaultValue = context.getResources().getBoolean(
-                R.bool.swipe_deletes_conversation_default);
+                R.bool.swipe_right_deletes_conversation_default);
         return prefs.getBoolean(prefKey, defaultValue);
     }
 }
-- 
2.17.1

